File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        freq = [[] for i in range(len(nums)+1)] #index 0 is unavailable
        count = {}
        output = []
        for num in nums:
            count[num] = 1+count.get(num,0)
        for num,cnt in count.items():
            freq[cnt].append(num)
        for i in range(len(freq)-1,0,-1):
            for value in freq[i]:
                output.append(value)
                if len(output) == k:
                    return output
        return output

File: Data Structures & Algorithms/partition-equal-subset-sum/submission-0.cpp
class Solution {
public:
    bool canPartition(vector<int>& nums) {
        int sum = 0;
        for(int num : nums){
            sum += num;
        }
        if(sum % 2 != 0){return false;}
        int target = sum /2;
        vector<bool> dp(target+1,false);
        dp[0] = true;
        for(int num : nums){
            for(int i = target; i >= num; i--){
                if(dp[i-num]){
                    dp[i] = true;
                }
                if(dp[target]){return true;}
            }
        }
        return false;
    }
};
